---
# return process angle
float32 angle